src/main/java/com/google/sps/DataServlet.java

```diff
@@ -164,37 +201,59 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       return;
     }
 
-    List<String> queried = new ArrayList<>();
-    if (nodeNameParam.length() > 0) {
-      queried.add(nodeNameParam);
-    }
-    // If the token is contained, then get the nodes associated with the token and
-    // add them to the queried nodes
+    // Show mutations relevant to nodes that contain the token in the new graph
     if (currDataGraph.tokenMap().containsKey(tokenParam)) {
       queried.addAll(currDataGraph.tokenMap().get(tokenParam));
+      queriedNext.addAll(currDataGraph.tokenMap().get(tokenParam));
     }
+
+    // Truncate the graph from the nodes that the client had searched for
     truncatedGraph = currDataGraph.getReachableNodes(queried, depthNumber);
 
-    if (nodeNameParam.length() == 0 && truncatedGraph.equals(currDataGraph.graph())) {
-      // If we are not filtering the graph or limiting its depth, show all mutations of all nodes
+    // To get the nodes to calculate relevant mutations from. If queried and queried next contain
+    // the same
+    // nodes, then no reason to regenerate the graph
+    MutableGraph<GraphNode> truncatedGraphNext =
+        queried.equals(queriedNext)
+            ? Graphs.copyOf(truncatedGraph)
```

Comment:
Is there a reason why we create a copy when queried is equal to queried next. I don't think we mutate either so there shouldn't be a need to copy the internal data structures?